discovery_engine_core/core/src/stack/ops.rs

```diff
@@ -50,6 +50,9 @@ pub trait Ops {
     /// tailored to the user's interests.
     async fn new_items(&self, key_phrases: &[KeyPhrase]) -> Result<Vec<Article>, GenericError>;
 
+    /// Returns if `[new_items]` needs the key phrases to work.
+    fn needs_key_phrases(&self) -> bool;
```

Comment:
if the returned value is supposed to be constant per impl anyways, why not make it an associated constant of the trait?

Reply:
Associated const are not object safe.

Reply:
i see, same old dilemma :/ not sure if i have asked this before, is there a reason we don't use static dispatch instead? like replacing `stack_ops: Vec<Box<dyn Ops>>` with `stack_ops: Vec<impl Ops>` or is there sth else limiting us?